platforms/pf_test/platform.cpp

```diff
@@ -249,6 +258,87 @@ static void *_task_planner_start_cancel(void *arg)
     return NULL;
 }
 
+/// Execute a shell command callback using arguments from Protobuf message.
+/// Command name is in the 'cmd' attribute of the Protobuf message.
+/// Arguments are in a space-separated string in 'desc' attribute of the Protobuf message.
+static void run_command(cogip::shell::Command *command, const cogip::shell::Command::PB_Message *pb_command)
+{
+    // Computed number of arguments
+    int argc = 0;
+    // List of arguments null-separated
+    char args[COMMAND_DESC_MAX_LENGTH];
+    // Array of pointers to each argument
+    char *argv[MAX_COMMAND_ARGS];
+    // First argument is the command
+    argv[argc++] = (char *)pb_command->cmd();
+    if (pb_command->get_desc().get_length() != 0) {
+        // If there are arguments to the command in the 'desc' attribute
+        size_t i;
+        // Copy first argument pointer to 'argv'
+        argv[argc++] = args;
+        for (i = 0; i < pb_command->get_desc().get_length(); i++) {
+            assert(i < COMMAND_DESC_MAX_LENGTH);
+            assert(argc < MAX_COMMAND_ARGS);
+            char c = pb_command->desc()[i];
+            // Copy each argument in 'args'
+            args[i] = c;
+            if (c == ' ') {
+                // Insert a null character between each argument
+                args[i] = '\0';
+                // Copy next argument pointer to 'argv'
+                argv[argc++] = args + i + 1;
+            }
+        }
+        // Add null-character afeter last argument
+        args[i] = '\0';
+    }
+    // Execute shell command callback
+    command->handler()(argc, argv);
+}
+
+// Handle a Protobuf command message
+static void handle_command(const cogip::shell::Command::PB_Message *pb_command)
+{
+    // Search the command in current menu command
+    for (auto command: *cogip::shell::current_menu) {
+        if (command->name() == pb_command->cmd()) {
+            run_command(command, pb_command);
+            return;
+        }
+    }
+
+    // If command was not found in current menu,
+    // search the command in global commands
+    for (auto command: cogip::shell::global_commands) {
+        if (command->name() == pb_command->cmd()) {
+            run_command(command, pb_command);
+            return;
+        }
+    }
+}
+
+// Read incoming Protobuf message and call the corresponding message handler
+void message_handler(uint8_t message_type, cogip::uartpb::ReadBuffer &buffer)
```

Comment:
This function is used where ? No prototype in a header ?

Reply:
It is a parameter of UartProtobuf constructor.